storage/remote/queue_manager.go

```diff
@@ -748,16 +752,17 @@ func (s *shards) sendSamplesWithBackoff(ctx context.Context, samples []prompb.Ti
 	}
 }
 
-func buildWriteRequest(samples []prompb.TimeSeries, buf []byte) ([]byte, int64, error) {
+func (s *shards) buildWriteRequest(samples []sample, buf []byte) ([]byte, int64, error) {
 	var highest int64
-	for _, ts := range samples {
-		// At the moment we only ever append a TimeSeries with a single sample in it.
-		if ts.Samples[0].Timestamp > highest {
-			highest = ts.Samples[0].Timestamp
+	req := s.writeRequests.get(len(samples))
+	defer s.writeRequests.put(req)
```

Comment:
defer causes an alloc. I'd avoid sync.Pool entirely, it has interesting interactions with GC and in this case we know exactly how much memory we need.

Reply:
:+1: Removed the pool.